cunumeric/deferred.py

```diff
@@ -1924,6 +1924,32 @@ def unique(self):
 
         return result
 
+    @auto_convert([1, 2])
+    def searchsorted(self, rhs, v, side="left"):
+
+        task = self.context.create_task(CuNumericOpCode.SEARCHSORTED)
+
+        is_left = side == "left"
+
+        if is_left:
+            self.fill(np.array(rhs.size, self.dtype))
+            task.add_reduction(self.base, ReductionOp.MIN)
+        else:
+            self.fill(np.array(0, self.dtype))
+            task.add_reduction(self.base, ReductionOp.MAX)
+
+        task.add_input(rhs.base)
+        task.add_input(v.base)
+
+        # every partition needs the value information
+        task.add_broadcast(v.base)
+        task.add_broadcast(self.base)
```

Comment:
with this approach we map entire `v` and `self` to a single thread/task. Wile for the case when `v` is an array I don't necessarily see how can we implement this differently, we can do an optimization for the case when v is a scalar: in this case we can partition `self` and broadcast `v`

Reply:
I don't really understand that. In case `v` is a scalar `self `should be a scalar as well, right? I just treat/convert scalars as ndarrays of size 1 for simplicity.